src/transaction/utils.ts

```diff
@@ -63,25 +64,35 @@ function prepareTransaction(txJSON: any, api: RippleAPI,
     const multiplier = instructions.signersCount === undefined ? 1 :
       instructions.signersCount + 1
     if (instructions.fee !== undefined) {
+      const fee = new BigNumber(instructions.fee)
+      if (fee.greaterThan('2') && instructions.allowHighFee !== true) {
+        const errorMessage = `Fee of ${fee.toString(10)} XRP exceeds ` +
+          'soft limit of 2 XRP. To use this fee, set ' +
+          '`instructions.allowHighFee = true`.'
+        throw new ValidationError(errorMessage)
+      }
       txJSON.Fee = scaleValue(common.xrpToDrops(instructions.fee), multiplier)
       return Promise.resolve(txJSON)
     }
     const cushion = api._feeCushion
-    return api.getFee(cushion).then(fee => {
+    return api.getFee(cushion, null).then(fee => {
       return api.connection.getFeeRef().then(feeRef => {
         const extraFee =
           (txJSON.TransactionType !== 'EscrowFinish' ||
             txJSON.Fulfillment === undefined) ? 0 :
             (cushion * feeRef * (32 + Math.floor(
               new Buffer(txJSON.Fulfillment, 'hex').length / 16)))
         const feeDrops = common.xrpToDrops(fee)
-        if (instructions.maxFee !== undefined) {
-          const maxFeeDrops = common.xrpToDrops(instructions.maxFee)
-          const normalFee = scaleValue(feeDrops, multiplier, extraFee)
-          txJSON.Fee = BigNumber.min(normalFee, maxFeeDrops).toString()
-        } else {
-          txJSON.Fee = scaleValue(feeDrops, multiplier, extraFee)
+
+        let maxFee = instructions.maxFee
```

Comment:
`const maxFee = instructions.maxFee || '2'` would allow you to define this as a const.